packages/server/src/integrations/base/clientCache.ts

```diff
@@ -0,0 +1,65 @@
+import { Duration } from "@budibase/backend-core"
+import LRUCache from "lru-cache"
+
+interface Merkle {
+  hash: (json: Record<string, any>) => string
+  stringify: (json: Record<string, any>) => string
+}
+
+interface MerkleConstructor {
+  new (): Merkle
+}
+
+// no typescript definition - fill it in
+const { MerkleJson } = require("merkle-json") as {
+  MerkleJson: MerkleConstructor
+}
+
+const jsonHash = new MerkleJson()
+
+type ClientCacheValue<T> = {
+  destroy: (client: T) => Promise<void>
+  // can't really know the type of the client, they are all so different
+  client: T
+}
+
+const clientCache = new LRUCache<string, ClientCacheValue<any>>({
+  // max 50 active clients per service
+  max: 50,
+  ttl: Duration.fromMinutes(10).toMs(),
```

Comment:
This is almost certainly higher than the time an idle connection will be kept open by the database server, so you may find that after 9 minutes a client goes to use it and you get a connection reset error. I'd probably be looking to keep this to 60 seconds or less.

Reply:
If we use a connection pool then this issue should go away!

Reply:
Pooling implemented.